fdb-record-layer-lucene/src/main/java/com/apple/foundationdb/record/lucene/LucenePartitioner.java

```diff
@@ -1070,15 +1088,41 @@ private CompletableFuture<Integer> moveDocsFromPartition(@Nonnull final LuceneRe
                 // create a new "overflow" partition
                 destinationPartition = newPartitionMetadata(overflowPartitioningKey, repartitioningContext.maxPartitionId + 1);
                 savePartitionMetadata(groupingKey, destinationPartition.toBuilder());
+                timings.createPartitionNanos = System.nanoTime();
             }
+            long updateStart = System.nanoTime();
 
             Iterator<? extends FDBIndexableRecord<Message>> recordIterator = records.iterator();
             final int destinationPartitionId = destinationPartition.getId();
             return AsyncUtil.whileTrue(() -> indexMaintainer.update(null, recordIterator.next(), destinationPartitionId)
                     .thenApply(ignored -> recordIterator.hasNext()), state.context.getExecutor())
                     .thenApply(ignored -> {
                         if (LOGGER.isDebugEnabled()) {
-                            LOGGER.debug(repartitionLogMessage("Repartitioned records", groupingKey, records.size(), partitionInfo));
+                            long updateNanos = System.nanoTime();
+                            final KeyValueLogMessage logMessage = repartitionLogMessage("Repartitioned records", groupingKey, records.size(), partitionInfo);
+                            logMessage.addKeyAndValue("totalMicros", TimeUnit.NANOSECONDS.toMicros(updateNanos - timings.startNanos));
+                            logMessage.addKeyAndValue("initializationMicros", TimeUnit.NANOSECONDS.toMicros(timings.initializationNanos - timings.startNanos));
+                            logMessage.addKeyAndValue("searchMicros", TimeUnit.NANOSECONDS.toMicros(timings.searchNanos - timings.initializationNanos));
+                            logMessage.addKeyAndValue("clearInfoMicros", TimeUnit.NANOSECONDS.toMicros(timings.clearInfoNanos - timings.searchNanos));
+                            if (timings.emptyingNanos > 0) {
+                                logMessage.addKeyAndValue("emptyingMicros", TimeUnit.NANOSECONDS.toMicros(timings.emptyingNanos - timings.clearInfoNanos));
+                            }
+                            if (timings.deleteNanos > 0) {
+                                logMessage.addKeyAndValue("deleteMicros", TimeUnit.NANOSECONDS.toMicros(timings.deleteNanos - timings.clearInfoNanos));
+                            }
+                            if (timings.metadataUpdateNanos > 0) {
+                                logMessage.addKeyAndValue("metadataUpdateMicros", TimeUnit.NANOSECONDS.toMicros(timings.metadataUpdateNanos - timings.deleteNanos));
```

Comment:
What if `timings.deleteNanos` == 0?

Reply:
Those are set in the same `if` block, so they will either both be set, or both unset.